tests/foreman/api/test_hostgroup.py

```diff
@@ -384,453 +308,223 @@ def test_positive_create_with_media(self):
 
-        :expectedresults: A hostgroup is updated with expected architecture
+        :expectedresults: A hostgroup is created with expected multiple
+            locations assigned
 
         :CaseLevel: Integration
         """
+        locs = [
+            entities.Location(organization=[self.org]).create()
+            for _ in range(randint(3, 5))
+        ]
         hostgroup = entities.HostGroup(
-            architecture=entities.Architecture().create(),
-            location=[self.loc],
+            location=locs,
             organization=[self.org],
         ).create()
-        new_arch = entities.Architecture().create()
-        hostgroup.architecture = new_arch
-        hostgroup = hostgroup.update(['architecture'])
-        self.assertEqual(hostgroup.architecture.read().name, new_arch.name)
+        self.assertEqual(
+            set(loc.name for loc in locs),
+            set(loc.read().name for loc in hostgroup.location)
+        )
 
     @tier2
-    def test_positive_update_media(self):
-        """Update a hostgroup with a new media
+    def test_positive_create_with_orgs(self):
+        """Create a hostgroup with multiple organizations specified
 
-        :id: 9b6ffbb8-0518-4900-95fd-49fc1d90a4be
+        :id: 09642238-cf0d-469a-a0b5-c167b1b8edf5
 
-        :expectedresults: A hostgroup is updated with expected media
+        :expectedresults: A hostgroup is created with expected multiple
+            organizations assigned
 
         :CaseLevel: Integration
         """
-        arch = entities.Architecture().create()
-        ptable = entities.PartitionTable().create()
-        os = entities.OperatingSystem(
-            architecture=[arch],
-            ptable=[ptable],
-        ).create()
-        media = entities.Media(
-            operatingsystem=[os],
-            location=[self.loc],
-            organization=[self.org],
-        ).create()
-        hostgroup = entities.HostGroup(
-            architecture=arch,
-            location=[self.loc],
-            medium=media,
-            operatingsystem=os,
-            organization=[self.org],
-            ptable=ptable,
-        ).create()
-        new_media = entities.Media(
-            operatingsystem=[os],
-            location=[self.loc],
-            organization=[self.org],
-        ).create()
-        hostgroup.medium = new_media
-        hostgroup = hostgroup.update(['medium'])
-        self.assertEqual(hostgroup.medium.read().name, new_media.name)
+        orgs = [
+            entities.Organization().create()
+            for _ in range(randint(3, 5))
+        ]
+        hostgroup = entities.HostGroup(organization=orgs).create()
+        self.assertEqual(
+            set(org.name for org in orgs),
+            set(org.read().name for org in hostgroup.organization)
+        )
 
-    @tier2
-    def test_positive_update_ptable(self):
-        """Update a hostgroup with a new partition table
+    @tier1
+    def test_positive_update_name(self):
```

Comment:
I'm fine if you add name as one of properties in `test_positive_create_with_properties` and just remove this test. Seems hostgroups never had any special requirements around names, so there is little value in running it.